src/test/java/org/alfresco/repo/event2/UpdateRepoEventIT.java

```diff
@@ -297,8 +302,239 @@ public void testCreateAndUpdateInTheSameTransaction()
+
+        retryingTransactionHelper.doInTransaction(() -> {
+            nodeService.moveNode(
+                moveFolder,
+                grandParent,
+                ContentModel.ASSOC_CONTAINS,
+                QName.createQName(TEST_NAMESPACE));
+            return null;
+        });
+
+        checkNumOfEvents(4);
+
+        RepoEventContainer repoEventsContainer = getRepoEventsContainer();
+
+        final String grandParentID = getNodeResource(repoEventsContainer.getEvent(1)).getId();
+        final String parentID = getNodeResource(repoEventsContainer.getEvent(2)).getId();
+
+        final String moveFolderParentBeforeMove =
+            getNodeResourceBefore(repoEventsContainer.getEvent(4)).getPrimaryHierarchy().get(0);
+        final String moveFolderParentAfterMove =
+            getNodeResource(repoEventsContainer.getEvent(4)).getPrimaryHierarchy().get(0);
+
+        assertTrue("Wrong parent.",parentID.equals(moveFolderParentBeforeMove));
+        assertTrue("Wrong parent.",grandParentID.equals(moveFolderParentAfterMove));
+        assertEquals("Wrong repo event type.", EventType.NODE_UPDATED.getType(),
+            getRepoEvent(4).getType());
+    }
+
+    @Test
+    public void testMoveFolderStructure()
+    {
+        final NodeRef root1 = createNode(ContentModel.TYPE_FOLDER);
+        final NodeRef root2 = createNode(ContentModel.TYPE_FOLDER);
+        final NodeRef grandParent = createNode(ContentModel.TYPE_FOLDER, root1);
+        final NodeRef parent = createNode(ContentModel.TYPE_FOLDER, grandParent);
+        createNode(ContentModel.TYPE_CONTENT, parent);
+
+        retryingTransactionHelper.doInTransaction(() -> {
+            nodeService.moveNode(
+                grandParent,
+                root2,
+                ContentModel.ASSOC_CONTAINS,
+                QName.createQName(TEST_NAMESPACE));
+            return null;
+        });
+
+        checkNumOfEvents(6);
+
+        RepoEventContainer repoEventsContainer = getRepoEventsContainer();
+
+        final String root2ID = getNodeResource(repoEventsContainer.getEvent(2)).getId();
+        final String grandParentParentAfterMove =
+            getNodeResource(repoEventsContainer.getEvent(6)).getPrimaryHierarchy().get(0);
+        assertTrue("Wrong parent.",root2ID.equals(grandParentParentAfterMove));
+
+        final String grandParentID = getNodeResource(repoEventsContainer.getEvent(3)).getId();
+        final String parentIDOfTheParentFolder =
+            getNodeResource(repoEventsContainer.getEvent(4)).getPrimaryHierarchy().get(0);
+        assertTrue("Wrong parent.",grandParentID.equals(parentIDOfTheParentFolder));
+
+        final String parentID = getNodeResource(repoEventsContainer.getEvent(4)).getId();
+        final String contentParentID =
+            getNodeResource(repoEventsContainer.getEvent(5)).getPrimaryHierarchy().get(0);
+        assertTrue("Wrong parent.",parentID.equals(contentParentID));
```

Comment:
Please use `assertEquals`
@aftene, rather than adding a comment for each `assertTrue("Wrong parent."...` in all the tests, please check the rest and modify them.